src/main/scala/chisel3/package.scala

```diff
@@ -436,6 +436,28 @@ package object chisel3 {    // scalastyle:ignore package.object.name
 
     val IO = chisel3.core.IO
 
+    // Rocket Chip-style clonemodule
+
+    /** A record containing the results of CloneAndGetIO.
+      * The apply method is retrieves the element with the supplied name.
+      */
+    type ClonePorts = chisel3.core.BaseModule.ClonePorts
+
+    object CloneAndGetIO {
+      /** Clones an existing module and returns a record of all its top-level ports.
+        * Each element of the record is named with a string matching the
+        * corresponding port's name and shares the port's type.
+        * @example {{{
+        * val q1 = Module(new Queue(UInt(32.W), 2))
+        * val q2_io = CloneAndGetIO(q1)("io").asInstanceOf[q1.io.type]
```

Comment:
Will this still work post-structural types?

Reply:
I'm not sure what part of the code you're referring to here.

Reply:
Will `q2_io` still have the right type or will it just fall back to `Bundle` post-structural types?

Reply:
As far as I can tell, I do not use structural types anywhere here, so this PR should be totally orthogonal to that issue. If someone writes a module that has an anonymous bundle that is accessed with inferred structural types and support for that feature is removed, it is true that it will not be possible to clone that module. However, that is because it will not be possible to even *compile* a circuit using that module.